### ISSUE TITLE
Replicate input structure in output files

### ISSUE THREAD

**Issue by angus-g** [angus-g/lagrangian-filtering#1]
The output file is simply a concatenation of the filtered, flattened arrays for each timestep:
```
HDF5 "test.h5" {
GROUP "/" {
   DATASET "var_U" {
      DATATYPE  H5T_IEEE_F32LE
      DATASPACE  SIMPLE { ( 2, 70000 ) / ( 2, 70000 ) }
   }
}
}
```

It would be good to copy the structure of the input file (spatial and temporal dimensions) to the output file.

**Comment by angus-g** [comment]
The most straightforward approach would probably be to open the xarray DataArray for each sampled variable, and combine them into a single Dataset. However, this interface doesn't support incremental writes. If we can set the Dataset's data to a lazy dask array, we could get around this?

**Comment by angus-g** [comment]
https://github.com/pydata/xarray/issues/1672 pertains to the feature in xarray that would make this easier.

**Comment by angus-g** [comment]
Closed via #14
